firefox_puppeteer/ui/windows.py

```diff
@@ -449,13 +450,20 @@ def is_private(self):
                 return PrivateBrowsingUtils.isWindowPrivate(chromeWindow);
             """, script_args=[self.window_element])
 
-    @use_class_as_property('ui.toolbars.NavBar')
+    @property
     def navbar(self):
         """Provides access to the navigation bar. This is the toolbar containing
         the back, forward and home buttons. It also contains the location bar.
 
         See the :class:`~ui.toolbars.NavBar` reference.
         """
+        self.switch_to()
+
+        if not self._navbar:
+            from .toolbars import NavBar
+            self._navbar = NavBar(lambda: self.marionette, self, self.window_element)
```

Comment:
The same here for the element to pass in. In this case it has to be the DOM node with the ID `nav-bar`. It's not used in the NavBar class yet, but will be available then via the `element` property.